dashboard/app/helpers/javalab_files_helper.rb

```diff
@@ -0,0 +1,49 @@
+
+module JavalabFilesHelper
+  # Get all files related to the project at the given channel id as a hash. The hash is in the format
+  # below. All values are StringIO.
+  # {
+  #   "main.json": <main source file for a project>
+  #   "assets": {"asset_name_1": <asset_value>, ...}
+  #   "validation": <all validation code for a project, in json format>
+  #   "maze": <serialized maze if it exists>
+  # }
+  # If the channel doesn't have validation and/or a maze, those fields will not be present.
+  def self.get_project_files(channel_id)
+    storage_id, storage_app_id = storage_decrypt_channel_id(channel_id)
+    all_files = {}
+    # get sources file
+    source_data = SourceBucket.new.get(channel_id, "main.json")
+    # Note: we can call .string on this value (and all other values for files) to get the raw string.
```

Comment:
Nice - for reference, I'm looking into zipping files in rails right now, and passing a string value for each file seems to be the easiest option, so this will work well!